### ISSUE TITLE
Issue with hibernation and wifi

### ISSUE THREAD

**Issue by alexmenn** [daliansky/XiaoMi-Pro#102]
Hello all!
I've bought a new xiaomi notebook pro 15, i7 with 16gb ram for my office.
Bios version: 603, I patched the scripts on w10 (fan scripts, speedshift, and so, like my other notebook pro).
I had to change the ssd pm981 to a samsung 970 m2 because of the incompatibility, and now everything is installed and working fine, except wifi and hibernation.
When I installed AirportBrcmFixup.kext to check the wifi, after restart, wifi worked ok. But after first hibernation, the wifi crash, and hibernation too. I mean, after the first sleep, the wifi didnt work anymore (I rebooted more than 30 times and tried everything without success), and the hibernation now doesnt work too. It goes to sleep ok, but when I press power button, the screen keeps black. I tried to replace full EFI folder again (newest release one) but nothing. I have dumped all files required by rehabman, maybe you can guide me.

The wifi chip I'm using is this: DW1560 BCM94352Z NGFF M.2 with shentek adapter in 2nd SSD slot, like my other notebook.

[debug_23685.zip](https://github.com/daliansky/XiaoMi-Pro/files/2579884/debug_23685.zip)

**Comment by stevezhengshiqi** [comment]
Hi, I am using BCM943602CS without any kexts like AirportBrcmFixup.kext, and it works great. I don't come up with an idea about your problem. Maybe you can leave the issue open here or do the `Disable Hibernation` part in https://www.tonymacx86.com/threads/guide-native-power-management-for-laptops.175801/.

**Comment by alexmenn** [comment]
I followed rehabman guide to disable hibernation but hibernation still "hibernates". When I click on sleep, notebook goes to sleep after few seconds. And wifi still doesnt work. I have bought 2 packs of BCM943602CS with adapter to ngff m2 (1 in aliexpress and 1 in taobao).

**Comment by stevezhengshiqi** [comment]
You can try to remove `HibernationFixup.kext`in `/CLOVER/kexts/Other`, check `NeverHiberante`, and uncheck `HibernationFixup` in `config.plist`.

**Comment by alexmenn** [comment]
OK. Then, with BCM943602CS, you can hibernate without troubles with wifi, right?

**Comment by Menchen** [comment]
@alexmenn so far mine is working like a charm. Highly recommended.

**Comment by alexmenn** [comment]
Yeah! The chip arrived today, works like a charm, really cool. Now I need 3 cool mhf4 antennas, who can give me aliexpress link of good antennas?

**Comment by kampav** [comment]
@alexmenn , can you please post some pictures, I bought the same chips and adapter mentioned in this post. They are still in flight. Also a link to procedure will be much appreciated.

Thanks

**Comment by alexmenn** [comment]
Hi @kampav . My notebook is closed, but the process is very easy. You just have to hold the chip and adapter with any tape and then hold the adapter to the m2 slot with the screw :)

I didnt weld the bluetooth wires yet, but I will.

**Comment by kampav** [comment]
@alexmenn , thanks mate, once you open the box for soldering. Please upload some pictures before and after solder. Thank you so much in advance.

**Comment by alexmenn** [comment]
hi @stevezhengshiqi . Im speaking with Rehabman (tonymacx expert) and we are having some troubles injecting 3rd party kext in probook.
You may consider a little change in config.plist to avoid kext injection in clover folder and in /S/L:
config.plist/SystemParameters/InjectKexts=Yes is wrong. Should be "Detect".

**Comment by stevezhengshiqi** [comment]
@alexmenn Thank you, I may consider that. It is set to Yes because I am afraid that some people may install kexts incorrectly, and this option can let system ignore the unsupported kexts. You can put your kexts into CLOVER folder as well.

**Comment by Alkanna** [comment]
Sadly experiencing the same issue with op's original card.
It works only with a full reboot somehow, when you click reboot it says hardware missing, when you click shutdown and press power button to boot it works.  Also if computer hibernates, wifi is stuck on disabled, BUT if you hibernate again it re enables itself !
